### DISCUSSION TITLE
Shared Network is not working when host Mac M1 Firewall is enabled

### DISCUSSION THREAD

**Discussion by ilyaresh** [utmapp/UTM#3530] [UTM Help (macOS)]
Running UTM 3.0.4 on Mac M1 OS 12.01.
If I have host Mac's Firewall enabled I'm not getting assigned ipv4 IP on the guest VMs (currently playing with centos 7 x86_64, but I think I also had it with arm ubuntu).
if Firewall (on Mac host) is enabled and you try to down/up the interface it will timeout.
![image](https://user-images.githubusercontent.com/8470672/150728781-58134fdf-8012-412f-b7e4-48d10a4dcb22.png)
when Firewall  (on Mac host) is disabled the interface comes up nicely

![image](https://user-images.githubusercontent.com/8470672/150728945-69ef3d41-29ab-47c3-a75e-5ba5df899d66.png)
and I have ipv4
![image](https://user-images.githubusercontent.com/8470672/150728973-ec393a73-0588-486c-a2f5-858d42cd7ffc.png)

I've tried to add UTM as "allowed" app like that, but it doesn't help 
![image](https://user-images.githubusercontent.com/8470672/150729040-ea4877e5-f340-4dd9-a24c-334da6eb649a.png)

**Comment by adespoton** [comment]
Interesting... I've been getting spotty acceptance of Shared Network across a number of my VMs, and you may have just helped me isolate the culprit: I think that Little Snitch may be to blame for the Shared Network interface failing to come up in some situations.  I'll add "look at how shared network is implemented" to my backlog of things to investigate :)  Possibly there's a sanity check it's doing that fails if certain packets are blocked.

**Comment by digitalqualms** [comment]
For a macOS 12.4 VM, I had to add `/System/Library/Frameworks/Virtualization.framework/Versions/A/XPCServices/com.apple.Virtualization.VirtualMachine.xpc/Contents/MacOS/com.apple.VirtualizationVirtualMachine` to the allowed firewall settings. Since it's inside an .xpc package, I had had to browse for it in Finder, "Show package contents" and then drag it to System Preferences.

Shared networking worked after that.

For other VM types, you might look at the process hierarchy in Activity Monitor under UTM to see if there are any other processes that might do the trick

**Reply by fulldecent** [reply]
For me in macOS Ventura 13.0, this option was not selectable:
<img width="767" alt="Screenshot 2022-12-28 at 10 40 50" src="https://user-images.githubusercontent.com/382183/209836723-20a676ae-9dcd-49f7-ac67-1de40a28392c.png">

**Reply by Issac-Lopez** [reply]
This is the workaround for this issue in Ventura that worked for me.

> For a macOS Ventura 13.2.1 (22D68), the following terminal command allowed me to change the Firewall settings.
> 
> `sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /System/Library/Frameworks/Virtualization.framework/Versions/A/XPCServices/com.apple.Virtualization.VirtualMachine.xpc/Contents/MacOS/com.apple.Virtualization.VirtualMachine`
> 
> Restarted MacBook, path was added to Firewall as, allows incoming connections. This solved an unreachable network issue that occurred while in Shared Network mode for all of my VMs.

**Reply by hraban** [reply]
Wow unbelievable. This works for me!

Tip if you can't select it (e.g. on Sonoma): after opening its containing folder (by dragging it to the file selector window from finder), you can _right click_ the item and select "Quick Look". This will open quick look; close that: now the item is selected and you can click "open".

Insanity.

**Comment by adespoton** [comment]
Ah; I use Little Snitch instead of the built in firewall, so have never run into this issue.

-Em

**Comment by alshdavid** [comment]
> I had had to browse for it in Finder, "Show package contents" and then drag it to System Preferences.

Seems like this isn't possible in Ventura as drag drop doesn't work with the Firewall settings. Is there a terminal command I can use?

**Reply by Issac-Lopez** [reply]
For a macOS Ventura 13.2.1 (22D68), the following terminal command allowed me to change the Firewall settings.

`sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /System/Library/Frameworks/Virtualization.framework/Versions/A/XPCServices/com.apple.Virtualization.VirtualMachine.xpc/Contents/MacOS/com.apple.Virtualization.VirtualMachine`

Restarted MacBook, path was added to Firewall as, allows incoming connections. This solved an unreachable network issue that occurred while in Shared Network mode for all of my VMs.

**Comment by virtu** [comment]
None of the above suggestions worked for me.

I added qemuhelper and qemulauncher in addition to UTM and `com.apple.VirtualizationVirtualMachine` to the firewall exceptions. My setup has been working for months, now, after a reboot, no more network in shared network mode. Bridge mode works, but it's not what I want.

**Reply by secangkirkopipanas** [reply]
Hi @virtu,

Your case is exactly the same as me. Now I'm using Bridge mode waiting for solution/workaround for Shared Network. Not really sure what's the root problem.

**Comment by tfgcampos** [comment]
I think I solved this by adding **/Applications/UTM.app** and **/Applications/UTM.app/Contents/XPCServices/QEMUHelper.xpc/Contents/MacOS/QEMULauncher.app** to firewall exceptions. To add the later, you need to drag and drop from Finder after clicking the Plus (+) icon in firewall settings.
Can you try?

**Reply by boogi** [reply]
Will add that I just tried this approach (and everything else mentioned above) and it didn't work. UTM 4.4.4(92). Distro "Debian (Rosetta)" (from UTM Gallery). Host is Mac M1, Sonoma 14.1.1.
Issue might be with Cisco VPN (that I can't disable as it's corporate laptop).
Bridge mode works fine.
